### PR TITLE
Stop using packDomainName in IsDomainName

### PR DESCRIPTION
**Pull request by tmthrgd**
I'm not sure if this is worth doing, but @miekg mentioned it in https://github.com/miekg/dns/pull/818#issuecomment-442355376. `IsDomainName` is documented as being incredibly liberal, so it may be possible to make it simpler at the expense of strictness.

### CODE REVIEW

File: defaults.go [hunk 1]
```diff
@@ -166,8 +167,68 @@ func (dns *Msg) IsEdns0() *OPT {
 // label fits in 63 characters, but there is no length check for the entire
 // string s. I.e.  a domain name longer than 255 characters is considered valid.
```
Comment (miekg) [inline defaults.go]: This comment is not true anymore

File: defaults.go [hunk 2]
```diff
@@ -166,8 +167,68 @@ func (dns *Msg) IsEdns0() *OPT {
...
+
+	// Each dot ends a segment of the name. Except for escaped dots (\.), which
+	// are normal dots.
+
+	var (
+		off    int
+		begin  int
+		wasDot bool
+	)
+	for i := 0; i < len(s); i++ {
```
Comment (miekg) [inline defaults.go]: Maybe refer to packDomainName and that this code is a (almost) copy of that routine there.

**Comment by miekg** [comment]
Think this will be a very welcome change. Not allocating a buffer here should a (huge?) win.

**Comment by tmthrgd** [comment]
I don't know if this will be a huge win performance wise as that buffer is only allocated when the name has escape sequences since #818. There will be a performance improvement when the name does have escape sequences though.

File: defaults.go (hunk 1 above)
Comment (tmthrgd) [reply defaults.go]: I actually don't think this was ever true.

**Pull request merged**